e2e/react/src/react-router.test.ts

```diff
@@ -65,4 +65,24 @@ describe('React Router Applications', () => {
     const buildResult = runCLI(`test ${appName}`);
     expect(buildResult).toContain('Successfully ran target test');
   });
+
+  it('should be able to test a react-router application with jest', async () => {
+    const appName = uniq('app');
+    runCLI(
+      `generate @nx/react:app ${appName} --use-react-router --routing --unit-test-runner=jest --no-interactive`
+    );
+
+    const buildResult = runCLI(`test ${appName}`);
+    expect(buildResult).toContain('Successfully ran target test');
+  });
+
+  it('should be able to type-check a react-router application', async () => {
+    const appName = uniq('app');
+    runCLI(
+      `generate @nx/react:app ${appName} --use-react-router --routing --no-interactive`
+    );
+
+    const buildResult = runCLI(`typecheck ${appName}`);
+    expect(buildResult).toContain('Successfully ran target typecheck');
+  });
```

Comment:
We could test both tasks in a single test case. It will provide a couple of benefits:

- Reuse the generated application and save time
- Confirm `typecheck` works when there are unit tests as well, so it would cover more things (the current second test case doesn't generate tests)